CMakeLists.txt

```diff
@@ -192,11 +192,12 @@ if(S2N_BLOCK_NONPORTABLE_OPTIMIZATIONS)
     target_compile_options(${PROJECT_NAME} PUBLIC -DS2N_BLOCK_NONPORTABLE_OPTIMIZATIONS=1)
 endif()
 
-target_compile_options(${PROJECT_NAME} PUBLIC -fPIC)
+target_compile_options(${PROJECT_NAME} PUBLIC -fPIC -include "${CMAKE_CURRENT_LIST_DIR}/utils/s2n_prelude.h")
```

Comment:
Should it be like this:
```suggestion
target_compile_options(${PROJECT_NAME} PUBLIC -fPIC)
target_compile_options(${PROJECT_NAME} PRIVATE -include "${CMAKE_CURRENT_LIST_DIR}/utils/s2n_prelude.h")
```

By making this PUBLIC, any project that depends, directly or indirectly, on S2N gets this compile option added to all their .c/.cpp files